src/vellum/workflows/nodes/displayable/final_output_node/node.py

```diff
@@ -78,12 +78,17 @@ def _validate_output_type_consistency(mcs, cls: Type) -> None:
                 if descriptor_type == declared_output_type:
                     type_mismatch = False
                     break
+
```

Comment:
going to followup this validation with the following cleanups:
- Move this method from the metaclass into the node class to get type help
- Move the validation from custom to using pydantic